digiwf-connector/digiwf-camunda-rest-connector-starter/src/main/java/de/muenchen/oss/digiwf/connector/adapter/camunda/rest/out/processdefinition/ProcessDefinitionAdapter.java

```diff
@@ -0,0 +1,23 @@
+package de.muenchen.oss.digiwf.connector.adapter.camunda.rest.out.processdefinition;
+
+import de.muenchen.oss.digiwf.connector.core.application.port.out.ProcessOutPort;
+import de.muenchen.oss.digiwf.connector.core.domain.ProcessDefinitionLoadingException;
+import feign.FeignException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ProcessDefinitionAdapter implements ProcessOutPort {
+
+    private final ProcessInstanceClient processInstanceClient;
+
+    @Override
+    public String loadProcessDefinition(final String processInstanceId) throws ProcessDefinitionLoadingException {
+        try {
+            return processInstanceClient.getRootProcessInstanceDetail(processInstanceId).getDefinitionName();
```

Comment:
@lmoesle was mir gerade aufgefallen ist, warum nutzt du den hier den Name und nicht den Key?

Reply:
Ich habe mich an der `getProcessInstanceDetail` Methode orientiert und das gleiche auch für die Root Instance hinzugefügt (https://github.com/it-at-m/digiwf-core/blob/dev/digiwf-engine/digiwf-engine-service/src/main/java/de/muenchen/oss/digiwf/process/instance/api/resource/ServiceInstanceController.java). Ich habe dann die `ServiceInstanceTO` Klasse wiederverwendet und die hatte keinen DefinitionKey, sondern nur den DefinitionName.

Mir ist aber beim Testen kein Unterschied zwischen dem DefinitionName und DefinitionKey aufgefallen, deswegen habe ich das so gemacht.

Reply:
In den realen Prozessen sind die Keys immer eindeutiger oft mit Prefix für alle zusammen gehörenden Prozesse und die Namen mit Leerzeichen und deutlich umfangreicher. Müsste man auf jeden Fall spätestens im Zuge der wirklichen Policies einbauen, aber ist für den Moment wahrscheinlich noch nicht so relevant.